subsys/bluetooth/nrf_bt_adv/providers/tx_power.c

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/sys/byteorder.h>
+
+#include <zephyr/bluetooth/hci.h>
+#include <zephyr/bluetooth/hci_vs.h>
+
+#include <bluetooth/nrf_bt_adv.h>
+
+#include <logging/log.h>
+LOG_MODULE_DECLARE(nrf_bt_adv, CONFIG_NRF_BT_ADV_LOG_LEVEL);
+
+
+static int get_data(struct bt_data *ad, const struct nrf_bt_adv_state *state,
+		    struct nrf_bt_adv_feedback *fb)
+{
+	ARG_UNUSED(state);
+	ARG_UNUSED(fb);
+
+	static uint8_t tx_power;
+
+	struct bt_hci_cp_vs_read_tx_power_level *cp;
+	struct bt_hci_rp_vs_read_tx_power_level *rp;
+	struct net_buf *rsp = NULL;
+	struct net_buf *cmd_buf = bt_hci_cmd_create(BT_HCI_OP_VS_READ_TX_POWER_LEVEL, sizeof(*cp));
+
+	if (!cmd_buf) {
+		return -ENOBUFS;
+	}
+
+	cp = net_buf_add(cmd_buf, sizeof(*cp));
+	cp->handle_type = BT_HCI_VS_LL_HANDLE_TYPE_ADV;
+	cp->handle = sys_cpu_to_le16(0);
+
+	int err = bt_hci_cmd_send_sync(BT_HCI_OP_VS_READ_TX_POWER_LEVEL, cmd_buf, &rsp);
+
+	if (err) {
+		uint8_t reason = rsp ?
+				((struct bt_hci_rp_vs_read_tx_power_level *) rsp->data)->status : 0;
+		LOG_ERR("Read Tx power err: %d reason 0x%02x", err, reason);
+	} else {
+		rp = (void *)rsp->data;
```

Comment:
Why not just `(struct bt_hci_rp_vs_read_tx_power_level *)`? It seems to be more clear to me

Reply:
Changed